src/BlockTypes/ProductSummary.php

```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
 	protected function register_block_type_assets() {
 		parent::register_block_type_assets();
 		$this->register_chunk_translations( [ $this->block_name ] );
+		return null;
+	}
+
+	/**
+	 * Register the context.
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		if ( ! empty( $content ) ) {
+			parent::register_block_type_assets();
+			$this->register_chunk_translations( [ $this->block_name ] );
+			return $content;
+		}
+
+		$post_id         = $block->context['postId'];
+		$product         = wc_get_product( $post_id );
+		$summary_content = self::get_summary_content( $product );
+
+		if ( ! $summary_content ) {
+			return null;
+		}
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+		$style              = isset( $classes_and_styles['styles'] ) ? ' style="' . $classes_and_styles['styles'] . '" ' : '';
+
+		$output  = '<div class="wc-block-components-product-summary ' . $classes_and_styles['classes'] . ' ' . $classname . '"';
+		$output .= $style;
+		$output .= '>';
+		$output .= '<p>' . $summary_content . '</p>';
+		$output .= '</div>';
+
+		return $output;
+	}
+
+	/**
+	 * Returns summary text for a product.
+	 *
+	 * @param [WC_Product] $product Product to get summary text for.
+	 * @return (string|null) Summary text for the product.
+	 */
+	protected function get_summary_content( $product ) {
+		$short_description = $product->get_short_description();
+		$summary_content   = $short_description ? $short_description : $product->get_description();
+
+		if ( ! $summary_content ) {
+			return null;
+		}
+
+		$summary = wc_trim_string( $summary_content, 150, '...' );
```

Comment:
[`...`  is the default suffix](https://github.com/woocommerce/woocommerce/blob/410f06b8c8098c9c6be8182c35257745c714308b/plugins/woocommerce/includes/wc-formatting-functions.php#L1082), so I would probably not pass it and rely on the default.
```suggestion
		$summary = wc_trim_string( $summary_content, 150 );
```

Reply:
Nice catch! I will remove it 🤝